lib/mp4/transmuxer.js

```diff
@@ -140,38 +214,102 @@ VideoSegmentStream = function(track) {
 
   delete track.minPTS;
 
-  this.push = function(data) {
-    collectDtsInfo(track, data);
+  this.getGopForFusion_ = function (nalUnit) {
```

Comment:
This function definitely needs a comment. I think it is worth thinking about a more descriptive name. "GOP fusion" sounds really cool but I would have a hard time guessing what that means. What about findClosestGop()?